File: order-service/src/main/java/com/ecommerce/order/service/AddressService.java
package com.ecommerce.order.service;

import com.ecommerce.order.dto.AddressDto;

import java.util.List;

public interface AddressService {

    List<AddressDto> getAllAddressesByUserId(Long userId);

    AddressDto getAddress(Long addressId);

    AddressDto createAddress(AddressDto addressDto);

    AddressDto updateAddress(Long addressId, AddressDto addressDto);

    void deleteAddress(Long addressId);

}

File: web_app/src/app/services/address.service.ts
import { HttpClient } from '@angular/common/http';
import { inject, Injectable } from '@angular/core';
import { Address } from '../models/order/address.model';
import { Observable } from 'rxjs';

@Injectable({
    providedIn: 'root'
})
export class AddressService {
    httpClient = inject(HttpClient);

    private baseUrl = 'http://localhost:8080/api/addresses';

    getUserAddresses(userId: number): Observable<Address[]> {
        return this.httpClient.get<Address[]>(`${this.baseUrl}/user/${userId}`);
    }

    getAddress(addressId: number): Observable<Address> {
        return this.httpClient.get<Address>(`${this.baseUrl}/${addressId}`);
    }

    createAddress(address: Address): Observable<Address> {
        return this.httpClient.post<Address>(`${this.baseUrl}`, address);
    }

    updateAddress(addressId: number, address: Address): Observable<Address> {
        return this.httpClient.put<Address>(`${this.baseUrl}/${addressId}`, address);
    }

    deleteAddress(addressId: number): Observable<void> {
        return this.httpClient.delete<void>(`${this.baseUrl}/${addressId}`);
    }
}
